### PR TITLE
chore: refactor docker-compose files

### PR DESCRIPTION
**Pull request by warrenv**
[PR Process](https://github.com/poetapp/documentation/blob/master/process/pr-review.md#pr-process) - [PR Review Checklist](https://github.com/poetapp/documentation/blob/master/process/pr-review.md#pr-review-checklist)

### Release
 Semantic release is enabled for this repository. Make sure you follow the right commit message convention. 
 We're using semantic-release's default — [Angular Commit Message Conventions](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines).

## Description of Changes

Fixes: #497 

This refactors the docker compose config files to reverse
an earlier commit that introduced `docker-compose.override.yml` into
the repo. This file is used to alter the service startup for customizing local development
needs so it should never be checked in.

- All services have been moved back into `docker-compose.yml`
- A unit test for load configuration was made more robust so that
  using environment variable overrides do not break it.
- bitcoind services were renamed to be more generic and which
  network they connect to was made configurable for better reuse.

### CODE REVIEW

File: docker-compose.override.yml.example [hunk 1]
```diff
@@ -1,12 +1,40 @@
 version: '3'
 
 services:
+  rabbitmq:
```
Comment (geoffturk) [inline docker-compose.override.yml.example]: Shouldn't it just be called `rabbit` like in `docker-compose.yml` below?

Comment (warrenv) [reply docker-compose.override.yml.example]: yes.

**Comment by warrenv** [comment]
I was going to leave testnet for a future refactor since I haven't dug into where it's being used yet.
Using env vars to choose test or regtest in this refactor is a move to potetially eliminating the testnet docker-compose.

File: docker-compose.yml [hunk 2]
```diff
@@ -8,22 +45,29 @@ services:
...
-      - BITCOIN_URL=bitcoin-regtest
-      - RABBITMQ_URL=amqp://poet-rabbit
-      - MONGODB_URL=mongodb://poet-mongo:27017/poet
-      - IPFS_URL=http://poet-ipfs:5001
+      - POET_SERVICE=node
+      - BITCOIN_URL=bitcoind-1
+      - RABBITMQ_URL=amqp://rabbit
+      - MONGODB_HOST=mongo
+      - IPFS_URL=http://ipfs:5001
       - INTEGRATION_TEST_MONGODB_URL=mongodb://poet-mongo:27017/poet
```
Comment (wzalazar) [inline docker-compose.yml]: Should be `mongodb://mongo:27017/poet` ?

Comment (warrenv) [reply docker-compose.yml]: It's not actually being used anywhere - I will remove them.

**Pull request merged**